src/ngraph/distributed.cpp

```diff
@@ -14,37 +14,66 @@
 // limitations under the License.
 //*****************************************************************************
 
-#ifdef NGRAPH_DISTRIBUTED
+#ifdef NGRAPH_DISTRIBUTED_ENABLE
 
+#ifdef NGRAPH_DISTRIBUTED_MLSL_ENABLE
 #include <mlsl.hpp>
+#else
+#include <mpi.h>
+#endif
 
 #include "ngraph/distributed.hpp"
 
 using namespace ngraph;
 
 ngraph::Distributed::Distributed()
 {
+#ifdef NGRAPH_DISTRIBUTED_MLSL_ENABLE
     if (!MLSL::Environment::GetEnv().IsInitialized())
     {
         MLSL::Environment::GetEnv().Init(nullptr, nullptr);
     }
+#else
+    int flag = 0;
+    MPI_Initialized(&flag);
+    if (!flag)
+    {
+        MPI_Init(NULL, NULL);
+    }
+#endif
 }
 
-ngraph::Distributed::~Distributed()
+void ngraph::Distributed::finalize()
```

Comment:
why don't keep it as a destructor? for RAII

Reply:
Addressed it for now, yet to see end to end integration